File: 06_Algorithm/leetcode_167.go
/**
 * File              : leetcode_167.go
 * Author            : Andy
 * Date              : 03.08.2022
 * Last Modified Date: 03.08.2022
 * Last Modified By  : Andy
 */

func twoSum(numbers []int, target int) int {
	l, r := 0, len(numbers)-1
	ans := make([]int, 2)
	for l < r {
		tmp := numbers[l] + numbers[r]
		if tmp > target {
			r --
		}else if tmp < target{
			l ++
		}else if tmp == target{
			ans[0], ans[1] := l+1, r+1
			return ans
		}

	}
	return []int{}
}

File: 06_Algorithm/leetcode_167.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# File              : leetcode_167.py
# Author            : Andy
# Date              : 03.08.2022
# Last Modified Date: 03.08.2022
# Last Modified By  : Andy


class Solution:
    def twoSum(self, numbers, target):
        left, right = 0, len(numbers) -1
        while left < right:
            tmp = numbers[left] + numbers[right]
            #结果大了，右边向左移
            #结果小了，左边右移
            if tmp > target:
                right -= 1
            elif tmp < target:
                left += 1
            elif tmp == target:
                return [left+1, right+1]
